src/main/java/com/opentable/versionedconfig/GitOperations.java

```diff
@@ -108,11 +108,17 @@ boolean pull() throws VersioningServiceException {
         LOG.trace("pulling latest");
         return upstreamRetry(remoteIndex -> {
             try {
+                int timeout = config.getTimeoutInSec();
                 final PullCommand pull = git.pull();
                 LOG.trace("Git pull completed");
                 configureCredentials(pull, config.getRemoteRepositories().get(remoteIndex));
                 LOG.trace("Configuration of credentials completed, setting remote {}", remoteIndex);
                 pull.setRemote("remote" + remoteIndex);
+
+                //set timeout if defined
+                if (timeout != -1){
+                    pull.setTimeout(timeout);
```

Comment:
worth a try I guess. IIRC Adam tried something simiolar and it didn't help. But heck, memory fades.

Reply:
Also as you say gitclone can probably use it. Could you please extend your PR to do so? I am assuming it's straightforward.